src/StreamJsonRpc.Shared/Reflection/TargetMethod.cs

```diff
@@ -28,12 +32,43 @@ internal TargetMethod(JsonRpcMessage request, object target, JsonSerializer json
             this.target = target;
 
             var targetMethods = new Dictionary<MethodSignature, object[]>();
+            bool errorFound = false;
             for (TypeInfo t = target.GetType().GetTypeInfo(); t != null; t = t.BaseType?.GetTypeInfo())
             {
-                bool matchFound = false;
-                foreach (MethodInfo method in t.GetDeclaredMethods(request.Method))
+                var matchFound = false;
+                var methodName = request.Method;
+                var methodAttribute = request.Method;
+
+                if (methodToAttributeMap != null)
+                {
+                    foreach (string methodKey in methodToAttributeMap.Keys)
+                    {
+                        if (methodToAttributeMap[methodKey].Any(a => string.Equals(a, request.Method, StringComparison.OrdinalIgnoreCase)))
+                        {
+                            if (methodToAttributeMap[methodKey].Count != 1)
+                            {
+                                errorFound = true;
+                                this.errorMessages.Add(string.Format(CultureInfo.CurrentCulture, Resources.ConflictingMethodNameAttribute, methodKey));
```

Comment:
When we first reflect over it it's in the ctor of JsonRpc, are you saying the construction should fail then?  I thought that might be overkill since maybe only one method has this error, but the rest are fine.